core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/impl/safe/AbstractNonDictionaryVectorFiller.java

```diff
@@ -0,0 +1,274 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.datastore.chunk.store.impl.safe;
+
+import java.nio.ByteBuffer;
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.metadata.datatype.DataType;
+import org.apache.carbondata.core.metadata.datatype.DataTypes;
+import org.apache.carbondata.core.scan.result.vector.CarbonColumnVector;
+import org.apache.carbondata.core.util.ByteUtil;
+import org.apache.carbondata.core.util.DataTypeUtil;
+
+public abstract class AbstractNonDictionaryVectorFiller {
+
+  protected int lengthSize;
+  protected int numberOfRows;
+
+  public AbstractNonDictionaryVectorFiller(int lengthSize, int numberOfRows) {
+    this.lengthSize = lengthSize;
+    this.numberOfRows = numberOfRows;
+  }
+
+  public abstract void fillVector(byte[] data, CarbonColumnVector vector, ByteBuffer buffer);
+
+  public int getLengthFromBuffer(ByteBuffer buffer) {
+    return buffer.getShort();
+  }
+}
+
+class NonDictionaryVectorFillerFactory {
+
+  public static AbstractNonDictionaryVectorFiller getVectorFiller(DataType type, int lengthSize,
+      int numberOfRows) {
+    if (type == DataTypes.STRING || type == DataTypes.VARCHAR) {
+      if (lengthSize == 2) {
+        return new StringVectorFiller(lengthSize, numberOfRows);
+      } else {
+        return new LongStringVectorFiller(lengthSize, numberOfRows);
+      }
+    } else if (type == DataTypes.TIMESTAMP) {
+      return new TimeStampVectorFiller(lengthSize, numberOfRows);
+    } else if (type == DataTypes.BOOLEAN) {
+      return new BooleanVectorFiller(lengthSize, numberOfRows);
+    } else if (type == DataTypes.SHORT) {
+      return new ShortVectorFiller(lengthSize, numberOfRows);
+    } else if (type == DataTypes.INT) {
+      return new IntVectorFiller(lengthSize, numberOfRows);
+    } else if (type == DataTypes.LONG) {
+      return new LongStringVectorFiller(lengthSize, numberOfRows);
+    }
+    return new StringVectorFiller(lengthSize, numberOfRows);
+  }
+
+}
+
+class StringVectorFiller extends AbstractNonDictionaryVectorFiller {
+
+  public StringVectorFiller(int lengthSize, int numberOfRows) {
+    super(lengthSize, numberOfRows);
+  }
+
+  @Override
+  public void fillVector(byte[] data, CarbonColumnVector vector, ByteBuffer buffer) {
+    // start position will be used to store the current data position
+    int startOffset = 0;
+    int currentOffset = lengthSize;
+    ByteUtil.UnsafeComparer comparer = ByteUtil.UnsafeComparer.INSTANCE;
+    for (int i = 0; i < numberOfRows - 1; i++) {
+      buffer.position(startOffset);
```

Comment:
```suggestion
(((data[offset] & 0xFF) << 8) | (data[offset + 1] & 0xFF));
```
based on above comment we can update this logic for all inner classes

Reply:
will add in another pr